opendebates/views.py

```diff
@@ -186,6 +186,20 @@ def vote(request, id):
         voter.source = request.COOKIES['opendebates.source']
         voter.save()
 
+    if voter.user and voter.user != request.user:
```

Comment:
The initial test (that voter.user is not None) is needed to allow unregistered users to vote. Unregistered users will have a Voter object with a `user` attribute set to `None`. Those voters will then be caught by the second test (because `AnonymousUser != None`) and will be disallowed from voting.